### PR TITLE
Improve RandomNumberGenerator docs.

### PR DESCRIPTION
**Pull request by ghost**
Clarified behavior of a few methods. Added implementation notes so that developers not familiar with RNGs are alerted to common pitfalls.

In response to https://github.com/godotengine/godot/issues/27094#issuecomment-478474145

### CODE REVIEW

File: doc/classes/RandomNumberGenerator.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 		A class for generation pseudo-random numbers.
 	</brief_description>
 	<description>
+		RandomNumberGenerator is a class for generation pseudo-random numbers. It currently uses PCG32. The underlying algorithm is an implementation detail. As a result, it should not be depended upon for reproducible random streams across Godot versions.
```
Comment (neikeq) [inline doc/classes/RandomNumberGenerator.xml]: Should be: `for generating pseudo-random numbers` or `for pseudo-random numbers generation`.

Comment (toasteater) [reply doc/classes/RandomNumberGenerator.xml]: True... I just copied `brief_description` without even reading it. Fixed both.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**